core-client/src/test/java/org/glassfish/jersey/client/ClientRxTest.java

```diff
@@ -147,27 +143,29 @@ public void testDefaultRxInvokerWithExecutorServiceProvider() throws ExecutionEx
                 .request().rx().get().toCompletableFuture().get()) {
 
             assertEquals(200, r.getStatus());
-            assertTrue("Executor Service was not passed to RxInvoker", threadName.get().contains("rxTest-"));
+            assertTrue(threadName.get().contains("rxTest-"), "Executor Service was not passed to RxInvoker");
         }
     }
 
     @Test
     public void testRxInvokerInvalid() {
-        Invocation.Builder request = target(CLIENT).request();
-        thrown.expect(IllegalArgumentException.class);
-        thrown.expectMessage(AllOf.allOf(new StringContains("null"), new StringContains("clazz")));
-        request.rx(null).get();
+        Assertions.assertThrows(IllegalArgumentException.class, () -> {
+            Invocation.Builder request = target(CLIENT).request();
+            // thrown.expectMessage(AllOf.allOf(new StringContains("null"), new StringContains("clazz")));
+            request.rx(null).get();
+        });
     }
 
     @Test
     public void testRxInvokerNotRegistered() {
-        Invocation.Builder request = target(CLIENT).request();
-        thrown.expect(IllegalStateException.class);
-        thrown.expectMessage(AllOf.allOf(
-                new StringContains("TestRxInvoker"),
-                new StringContains("not registered"),
-                new StringContains("RxInvokerProvider")));
-        request.rx(TestRxInvoker.class).get();
+        Assertions.assertThrows(IllegalStateException.class, () -> {
```

Comment:
JUnit 5 has support-class [ExpectedExceptionSupport](https://junit.org/junit5/docs/5.3.0/api/org/junit/jupiter/migrationsupport/rules/ExpectedExceptionSupport.html) for usage JUnit 4 Rule [ExpectedException](https://junit.org/junit4/javadoc/4.12/org/junit/rules/ExpectedException.html), but this support class also recomments to switch to assertThrows where possible ([source](https://junit.org/junit5/docs/5.3.0/api/org/junit/jupiter/migrationsupport/rules/ExpectedExceptionSupport.html)):
```
However, you should rather switch to [Assertions.assertThrows(java.lang.Class<T>, org.junit.jupiter.api.function.Executable)](https://junit.org/junit5/docs/5.3.0/api/org/junit/jupiter/api/Assertions.html#assertThrows(java.lang.Class,org.junit.jupiter.api.function.Executable)) for new code.
```
Of course I can restore usage of obsolete Rule,if you think it will better.

Reply:
my idea actually was just to a bit extend of what you are offering:
```
final IllegalStateException exception = Assertions.assertThrows(IllegalStateException.class, () -> {
            Invocation.Builder request = target(CLIENT).request();
            request.rx(TestRxInvoker.class).get();
        });
      
        final String message = exception.getMessage();
        assertTrue(AllOf.allOf(
                new StringContains("TestRxInvoker"),
                new StringContains("not registered"),
                new StringContains("RxInvokerProvider"))
                .matches(message)
        );
```
or something like that. I did not check my code, so the test might be failing as is but this is just an example.

Reply:
Yes, aou are right. This check was lost during migration.
Restored!